modules/generator/storage/config_util.go

```diff
@@ -33,7 +33,9 @@ func generateTenantRemoteWriteConfigs(originalCfgs []prometheus_config.RemoteWri
 				}
 			}
 
-			cloneCfg.Headers[user.OrgIDHeaderName] = tenant
+			if addOrgIDHeader {
```

Comment:
I'm wondering if we should move this if-case higher and merge it with `if tenant != util.FakeTenantID` at line 24.
Make it:

```go
if tenant != util.FakeTenantID && addOrgIDHeader {
```

Reasoning: we only need to remove any pre-existing X-Scope-OrgID headers if we want to inject our own. If we are not injecting any, there won't be a conflict and we don't need to remove them either.

We have the following possible scenarios:

- single-tenant: we don't remove X-Scope-OrgID
- multi-tenant with `addOrgIDHeader = true`: we remove X-Scope-OrgID headers and inject our own
- multi-tenant with `addOrgIDHeader = false`: we do not inject our own X-Scope-OrgID header so I don't think we need to remove them as well

Reply:
@kvrhdn I agree with your point of view. addOrgIDHeader = false make metrics generator more lenient on remote write authentication then it should also allow X-Scope-OrgID that a user wants to pass using remote write config. The suggested changes has been committed 364bdc39351837042c9c37b0430dfd091f4fc493